### DISCUSSION TITLE
Schouten identity simplifications

### DISCUSSION THREAD

**Discussion by gekatsia** [FeynCalc/feyncalc#181] [Q&A]
Hello,

I have a problem with the calculation of the parity odd terms of the traces of complicated expressions that contain gamma matrices.   It is quite often that after computing the trace of some  gamma matrices  Mathematica   ends up with some really huge  expressions(without being able to simplify them), which however, I know that they are supposed to be zero (or at least quite simpler). For example, please take a look at the following simple expression that I got after some computations.

`(Eps[LorentzIndex[b], LorentzIndex[k], Momentum[q1], 
     Momentum[q2]] Pair[LorentzIndex[a], LorentzIndex[j]] - 
   Eps[LorentzIndex[b], LorentzIndex[j], Momentum[q1], 
     Momentum[q2]] Pair[LorentzIndex[a], LorentzIndex[k]] + 
   Eps[LorentzIndex[b], LorentzIndex[j], LorentzIndex[k], 
     Momentum[q2]] Pair[LorentzIndex[a], Momentum[q1]] - 
   Eps[LorentzIndex[b], LorentzIndex[j], LorentzIndex[k], 
     Momentum[q1]] Pair[LorentzIndex[a], Momentum[q2]] - 
   Eps[LorentzIndex[a], LorentzIndex[k], Momentum[q1], 
     Momentum[q2]] Pair[LorentzIndex[b], LorentzIndex[j]] + 
   Eps[LorentzIndex[a], LorentzIndex[j], Momentum[q1], 
     Momentum[q2]] Pair[LorentzIndex[b], LorentzIndex[k]] - 
   Eps[LorentzIndex[a], LorentzIndex[j], LorentzIndex[k], 
     Momentum[q2]] Pair[LorentzIndex[b], Momentum[q1]] + 
   Eps[LorentzIndex[a], LorentzIndex[j], LorentzIndex[k], 
     Momentum[q1]] Pair[LorentzIndex[b], Momentum[q2]]) Pair[
  Momentum[q1], Momentum[q2]]`

where I have also used the assumption `SP[q1,q1]=0` and  `SP[q2,q2]=0`.
Using the Schouten identity to the appropriate indices, one can show by hand that the above expression evaluates to zero. 
I am using the //Schouten command but nothing happens.  I suppose the problem is that there are more than 5 indices and Mathematica does not know where to do the Schouten identity. Is there any way to show that the above is equal to zero?
Thank you for your help and time.
Best 
George

**Comment by vsht** [comment]
Hi,

you need [FCSchoutenBruteForce](https://feyncalc.github.io/FeynCalcBookDev/FCSchoutenBruteForce.html) for that. But is still involves a lot of trial and error to show that the given expression is zero by Schouten. 

Often the brute force procedure gets stuck in the sense, that there is no single replacement possible that would decrease the number of terms. Surprisingly, in such cases it often helps to make a replacement that would actually make the given expression larger. But then you are out of the pit and  the next replacement is most likely to be a good one, i.e. making the expression much shorter.

```
exp = SP[q1, q2] (MT[b, k] LC[a, j][q1, q2] - MT[b, j] LC[a, k][q1, q2] - 
   MT[a, k] LC[b, j][q1, q2] + MT[a, j] LC[b, k][q1, q2] + 
   FV[q2, b] LC[a, j, k][q1] - FV[q1, b] LC[a, j, k][q2] - 
   FV[q2, a] LC[b, j, k][q1] + FV[q1, a] LC[b, j, k][q2])
```

```
res1 = FCSchoutenBruteForce[exp, {}, {}, 
  SchoutenAllowNegativeGain -> True, SchoutenAllowZeroGain -> True]

FixedPoint[FCSchoutenBruteForce[#, {}, {}] &, res1]
(*0*)
```

In general, if the spurious zeros get too large, I would consider switching to FORM, where the trace function is more clever
in terms of avoiding such situations.

**Reply by gekatsia** [reply]
Thanks a lot!
